fix(curriculum): Updated step 74 of spreadsheet project and added step 75 and step 76 (#54417)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-functional-programming-by-building-a-spreadsheet/661f48f412d7631a1d9c30e6.md

Reviewer: The sei-colons here should be optional

Reviewer: Also it's possible for the regex tests to not be this complicated

---
id: 661f48f412d7631a1d9c30e6
title: Step 75
challengeType: 0
dashedName: step-75
---

# --description--

> You should use `console.log()` to return the `highPrecedence` function with the string `"5*3"`.

Reviewer 2: ```suggestion
You should use `console.log()` to print the result of calling the `highPrecedence` function with the string `"5*3"`.
```

`console.log()` returns `void`.

# --hints--

Please return the function using `console.log(highPrecedence("5*3"));`.

```js

assert.match(code, /console\.log\(highPrecedence\("5\*3"\)\);/);

> ```

Reviewer 2: We should make this more granular.

- Test for a function call of `highPrecedence`
- Test that `5*3` is passed as the argument
- Test that the result is logged.

---
id: 661f48f412d7631a1d9c30e6
title: Step 75
challengeType: 0
dashedName: step-75
---

# --description--

You should use `console.log()` to print the result of calling the `highPrecedence` function with the string `"5*3"`.

# --hints--

> Print or log the results using `console.log()`.

Reviewer 2: ```suggestion
You should call `console.log()`.
```

```js

> assert.match(code, /console\.log\(.*\);/);

Reviewer: This works fine: 
```suggestion
assert.match(code, /console\.log\(/);
```

```

> Make a test call for the `highPrecedence` function.

Reviewer 2: ````suggestion
You should call your `highPrecedence` function.

```js
assert.match(code, /highPrecedence\(.*\)/);
```

You should pass your `highPrecedence` call to your `console.log()` call.
````

```js

> 
> assert.match(code, /console\.log\(highPrecedence\((?:"[^"]*"|[^)]*)\)\);/);
> 
> 

Reviewer: This works fine too:
```suggestion
assert.match(code, /console\.log\(highPrecedence\(/);
```

```

Pass `5*3` as the argument

```js

> 
> assert.match(code, /console\.log\(highPrecedence\("5\*3"\)\);/);
> 

Reviewer: And this works fine:
```suggestion
assert.match(code, /console\.log\(highPrecedence\(["']5\*3["']\)\);?/);
```

We should allow single quotes, and make the semi-colon optional.
